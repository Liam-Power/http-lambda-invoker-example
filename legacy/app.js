const http = require('http');

const server = http.createServer((_, res) => {
  res.writeHead(200, { 'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*' });
  res.write(Buffer.from(JSON.stringify({ data: 'contains valuable legacy container data' })));
  res.end();
});
server.on('clientError', (err, socket) => {
  socket.end('HTTP/1.1 400 Bad Request\r\n\r\n');
});
server.listen(8000);
