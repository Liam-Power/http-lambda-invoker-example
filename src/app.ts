import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';

export const HttpLambdaInvokerExample = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
  console.log(event);
  return { body: JSON.stringify({ data: 'contains valuable serverless data' }), statusCode: 200 };
};
