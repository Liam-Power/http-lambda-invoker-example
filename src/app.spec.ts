import { APIGatewayProxyEvent } from 'aws-lambda';

import { HttpLambdaInvokerExample } from './app';

describe('default tests', () => {
  it('should be replaced with a real test', async () => {
    const response = await HttpLambdaInvokerExample({ path: '/data' } as APIGatewayProxyEvent);
    expect(response).toMatchSnapshot();
  });
});
