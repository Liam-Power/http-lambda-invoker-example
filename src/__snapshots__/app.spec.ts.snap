// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default tests should be replaced with a real test 1`] = `
Object {
  "body": "{\\"data\\":\\"contains valueable serverless data\\"}",
  "statusCode": 200,
}
`;
